### ISSUE TITLE
Chat freezes Habitica Website & Chat 

### ISSUE THREAD

**Issue by cTheDragons** [HabitRPG/habitica-chat-extension#22]
## General Info

User Id: bc10ec41-4213-4482-bbb8-558e3c3ea8d5
Opera 43.0.2442.1144
Windows 10 32-bit

*There are reports of this occurring on chrome browsers*

## Description
Often when opening large guilds like the Tavern, Habitica Help or Report a Bug the chat system freezes. You are unable to interact with either the Chat or Habitica website. Often you are unable to close the tab, and your only options is to kill the browser process.

When it freezes it appears to have loaded the chat correctly, however the Avatar images appear to be still loading. (Only previous avatars appear from previous loaded chat windows). 

![chaterror](https://cloud.githubusercontent.com/assets/14116069/23820496/ca24846c-066d-11e7-8247-2b9d1050e3ad.png)

Sometimes,  the Avatar images loads successfully, but the icon fails to change from finger pointing to arrow, to allow you to interact with the site. In this instance, you are able to close the browser.

## jConsole Error messages
Most of the time there is no-console error messages when the system freezes however occasionally the following messages show.

```
app-2b1449e3.js:5 GET https://habitica.com/api/v3/members/1ca11363-6e98-41b3-9bab-ecd08a78ea63 404 (Not Found)
send @ app-2b1449e3.js:5
ajax @ app-2b1449e3.js:5
lookUpMember @ chat_inPage.js:358
generateAvatar @ chat_inPage.js:399
digestChatData @ chat_inPage.js:305
success @ chat_inPage.js:178
j @ app-2b1449e3.js:3
fireWith @ app-2b1449e3.js:3
w @ app-2b1449e3.js:5
(anonymous) @ app-2b1449e3.js:5
app-2b1449e3.js:5 GET https://habitica.com/api/v3/members/21612986-ae65-4081-b485-ba1f01004104 404 (Not Found)
send @ app-2b1449e3.js:5
ajax @ app-2b1449e3.js:5
lookUpMember @ chat_inPage.js:358
generateAvatar @ chat_inPage.js:399
digestChatData @ chat_inPage.js:305
success @ chat_inPage.js:178
j @ app-2b1449e3.js:3
fireWith @ app-2b1449e3.js:3
w @ app-2b1449e3.js:5
(anonymous) @ app-2b1449e3.js:5
app-2b1449e3.js:5 GET https://habitica.com/api/v3/members/804a1937-b2eb-4e3f-8d1e-0c0c7e5f6a94 404 (Not Found)
send @ app-2b1449e3.js:5
ajax @ app-2b1449e3.js:5
lookUpMember @ chat_inPage.js:358
generateAvatar @ chat_inPage.js:399
digestChatData @ chat_inPage.js:305
success @ chat_inPage.js:178
j @ app-2b1449e3.js:3
fireWith @ app-2b1449e3.js:3
w @ app-2b1449e3.js:5
(anonymous) @ app-2b1449e3.js:5
app-2b1449e3.js:5 GET https://habitica.com/api/v3/members/cd581a7c-69d6-4f8c-a3b8-3f64e87452b7 404 (Not Found)
send @ app-2b1449e3.js:5
ajax @ app-2b1449e3.js:5
lookUpMember @ chat_inPage.js:358
generateAvatar @ chat_inPage.js:399
digestChatData @ chat_inPage.js:305
success @ chat_inPage.js:178
j @ app-2b1449e3.js:3
fireWith @ app-2b1449e3.js:3
w @ app-2b1449e3.js:5
(anonymous) @ app-2b1449e3.js:5
app-2b1449e3.js:5 GET https://habitica.com/api/v3/members/2b8a3412-fc34-4a11-b58b-154efd924e9e 404 (Not Found)
````

**Comment by UnexpectedMushroom** [comment]
I did a bit of experimenting to find out the smallest guild I'm in that this affects, and found something interesting. If you open up the small guilds first and let them load, then move progressively up the list, eventually you are able to open up the Newbies guild without a problem. When I opened the tavern after that it hung for long enough that chrome decided it was dead, but clicking "wait" once was enough to let it load.
After that I refreshed the browser and tried opening the tavern first, and it froze everything to the point where task manager was necessary. I haven't quite pinned down the minimum, but I know the back corner is big enough to crash it if it's the first one opened.

**Comment by Alys** [comment]
I wonder if that's because when you start with the small guilds, the extension loads few avatar images, and by the time you've worked up to the big guilds, enough images have been cached that you're still only loading a relatively small number for each guild. When you start with a big guild with no cache, you have to load many at once, causing it to hang. That's a guess though; I don't know if it does caching.

Does it have an option to not display avatar images? If so, you could try turning the images off, then reloading the page, then starting with a large guild.

**Comment by UnexpectedMushroom** [comment]
Nope, the only options are sound on/off and large text.

**Comment by crookedneighbor** [comment]
As a temporary stop gap, I've added a disable avatars option. The real solution will be to add some logic to limit the number of concurrent requests to avatars.

**Comment by cTheDragons** [comment]
The chat client is so fast now! I have not had any issues since this feature has been introduced.

Is it possible to leave this as permanent option? I still like to see the Avatar's pictures as it helps when I'm socialiting (helping those in the Habitica Help Guild & Tavern).

**Comment by cTheDragons** [comment]
This can be closed now as it has been fixed with Pull Request #37.
